pynumaflow/function/_dtypes.py

```diff
@@ -64,20 +58,78 @@ def items(self) -> List[Message]:
         return self._messages
 
     @classmethod
-    def as_forward_all(cls: Type[Ms], json_data: str) -> Ms:
+    def as_forward_all(cls: Type[Ms], value: bytes) -> Ms:
         msgs = cls()
-        if json_data:
-            msgs.append(Message.to_all(value=json_data.encode()))
+        if value:
+            msgs.append(Message.to_all(value=value))
         else:
             msgs.append(Message.to_drop())
         return msgs
 
-    def dumps(self, udf_content_type: str) -> str:
-        if udf_content_type == APPLICATION_JSON:
-            return json.dumps(self._messages, cls=NumaflowJSONEncoder, separators=(",", ":"))
-        elif udf_content_type == APPLICATION_MSG_PACK:
-            return msgpack.dumps(self._messages, default=msgpack_encoding)
-        raise MarshalError(udf_content_type)
+    def dumps(self) -> str:
+        return self.__str__()
 
     def loads(self) -> Ms:
         pass
+
+
+class Datum:
+    def __init__(self, value: bytes, event_time: datetime, watermark: datetime):
+        self._value = value or b""
+        if type(event_time) != datetime:
+            raise TypeError("wrong data type for Datum.event_time")
+        self._event_time = event_time
+        if type(watermark) != datetime:
+            raise TypeError("wrong data type for Datum.watermark")
+        self._watermark = watermark
+
+    def __str__(self):
+        return "value:%s event_time:%s watermark:%s" % (
+            self._value.decode("utf-8"),
+            str(self._event_time),
+            str(self._watermark),
+        )
+
+    def __repr__(self):
+        return str(self)
+
+    def value(self):
+        return self._value
+
+    def event_time(self):
+        return self._event_time
+
+    def watermark(self):
+        return self._watermark
+
+
+class IntervalWindow:
+    def __init__(self, start: datetime, end: datetime):
+        self._start = start
+        self._end = end
+
+    def __str__(self):
+        return "start:%s end:%s" % (str(self._start), str(self._end))
+
+    def __repr__(self):
+        return str(self)
+
+    def start(self):
+        return self._start
+
+    def end(self):
+        return self._end
```

Comment:
Make them as property
```suggestion
    @property
    def start(self):
        return self._start

    @property
    def end(self):
        return self._end
```